racetrack_client/racetrack_client/plugin/install.py

```diff
@@ -67,6 +69,62 @@ def uninstall_plugin(
     logger.info(f'Plugin {plugin_name} {plugin_version} has been uninstalled from {lifecycle_url}')
 
 
+def download_installed_plugin_version(
+        lifecycle_url: Optional[str],
+        out_dir: Optional[str | Path],
+        out_filename: Optional[str | Path ],
+        plugin_name: str,
+        plugin_version: str,):
+    """Download a plugin from a remote Racetrack server"""
+    client_config = load_client_config()
+    lifecycle_url = resolve_lifecycle_url(client_config, lifecycle_url)
+    user_auth = get_user_auth(client_config, lifecycle_url)
+
+    r = Requests.get(
+        f'{lifecycle_url}/api/v1/plugin/{plugin_name}/{plugin_version}/download',
+        headers=get_auth_request_headers(user_auth),
+    )
+    out_filename = f'{plugin_name}-{plugin_version}.zip'
+    _save_response_to_disk(r, out_dir, out_filename)
+
+
+def download_installed_plugins(lifecycle_url: Optional[str], out_dir: Optional[str | Path], out_filename: Optional[str | Path ]):
+    """Download all plugins from a remote Racetrack server"""
+    client_config = load_client_config()
+    lifecycle_url = resolve_lifecycle_url(client_config, lifecycle_url)
+    user_auth = get_user_auth(client_config, lifecycle_url)
+    r = Requests.get(
+        f'{lifecycle_url}/api/v1/plugin/download_installed_plugins',
+        headers=get_auth_request_headers(user_auth),
+    )
+    _save_response_to_disk(r, out_dir, out_filename)
+
+
+def _save_response_to_disk(r: Response, out_dir: Optional[str | Path], out_filename: Optional[str | Path ]):
```

Comment:
This function could be a lot simpler, if it would just abandon the idea of downloading all plugins.